src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/ProjectContextInfo.cs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft;
+using NuGet.Packaging;
+using NuGet.ProjectManagement;
+using NuGet.ProjectManagement.Projects;
+using NuGet.ProjectModel;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public class ProjectContextInfo
+    {
+        private const string LiveShareUriScheme = "vsls";
+        private const string ProjectGuidQueryString = "projectGuid";
+        private readonly Guid _projectUniqueId;
+
+        private ProjectContextInfo(Guid projectUniqueId)
+        {
+            _projectUniqueId = projectUniqueId;
+        }
+
+        public ProjectStyle ProjectStyle { get; private set; } = ProjectStyle.Unknown;
+        public NuGetProjectKind ProjectKind { get; private set; } = NuGetProjectKind.Unknown;
```

Comment:
If you fix [this issue](https://github.com/NuGet/NuGet.Client/pull/3485/files#diff-2b1025d2a870bf4c2c4986850fbde47bR14), default initialization is sufficient.